dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeAPIImpl.java

```diff
@@ -556,6 +564,40 @@ public ContentType copyFrom(final CopyContentTypeBean copyContentTypeBean) throw
     return newContentType;
   }
 
+  /**
+   * This method will generate a new relationship field based on the source field
+   * @param sourceField the source field
+   *
+   */
+  private Field generateNewRelationshipFiled(final Field sourceField) {
+
+    if (sourceField instanceof RelationshipField) {
```

Comment:
I would extract this `if (sourceField instanceof RelationshipField) {`  one level up
and call this method only if we're looking at a field of type RelationshipField. 
or rename this method to something like `generateNewRelationshipFiledIfNeeded`

Reply:
Perfect